src/rime/context.cc

```diff
@@ -123,6 +123,19 @@ bool Context::Select(size_t index) {
   return false;
 }
 
+bool Context::Delete(size_t index) {
```

Comment:
The name is unclear; it can be confused with deleting characters from the input.
Rename it `DeleteCandidate`.

Avoid repeated code.
Introduce a private member function (overloaded)
```c++
bool Context::DeleteCandidate(function<an<Candidate> (Segment& seg)> get_candidate) {
  // ...
  if (auto cand = get_candidate(seg)) {
  // ...
}
```
Then call it in each version:
```c++
bool Context::DeleteCandidate(size_t index) {
  return DeleteCandidate(
      [index](Segment& seg) {
        return seg.GetCandidateAt(index);
      });
}

bool Context::DeleteCurrentSelection() {
  return DeleteCandidate(
      [](Segment& seg) {
        return seg.GetSelectedCandidate();
      });
}
```